src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +41,87 @@ public static void main(String[] args) throws IOException {
+
+                    ProcessHandle externalProcessHandle = ProcessHandle.of(filePid).orElse(null);
+                    long externalStartTime = 0;
+                    if (externalProcessHandle != null) {
+                        ProcessHandle.Info ExternalProcessInfo = externalProcessHandle.info();
+                        externalStartTime = ExternalProcessInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+                    }
+
+                    if (externalProcessHandle != null && externalStartTime == fileStartTime) {
+                        JButton proceed = new JButton("Proceed Anyways", UIUtils.getIcon("diagnostics"));
+                        JButton cancel = new JButton("Cancel");
+                        AtomicInteger result = new AtomicInteger(-1);
+
+                        proceed.addActionListener(a -> {
+                            SwingUtilities.getWindowAncestor(proceed).setVisible(false);
+                            result.set(0);
+                        });
+                        cancel.addActionListener(a -> {
+                            SwingUtilities.getWindowAncestor(cancel).setVisible(false);
+                            result.set(1);
+                        });
+
+                        Popups.of("Multiple bots on the same folder",
+                                        "You're currently running multiple bot instances from the same folder.\n" +
+                                                "This can cause crash, unexpected problems, and some broken functionality.\n" +
+                                                "Please create a separate folder to run multiple instances.",
+                                        JOptionPane.WARNING_MESSAGE)
+                                .options(new Object[]{proceed, cancel})
+                                .initialValue(cancel)
+                                .showOptionSync();
+
+                        if (result.get() == 1) {
+                            System.out.println("Reject multiple bot warning, closing bot");
+                            System.exit(0);
+                        } else {
+                            Files.writeString(filePath, "\n" + currentPid + " " + currentStartTime, StandardOpenOption.APPEND);
+                        }
+                        break;
+                    } else {
+                        if (count == fileContent.size() - 1)
+                            Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.TRUNCATE_EXISTING);
+                    }
+                } catch (java.lang.NumberFormatException e) {
+                    e.printStackTrace();
+                    Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.TRUNCATE_EXISTING);
+                }
+                count++;
+            }
+
+        } else {
+            Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.CREATE);
+        }
```

Comment:
```suggestion
        Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
```
You do not need the if-s at all, you can simply write at the very end, and always write if it happens to reach here, if you decide to close it will exit earlier